cmd/main.go

```diff
@@ -555,12 +564,23 @@ func main() {
 		}
 	}()
 
-	address := fmt.Sprintf(":%s", swag.StringValue(port))
-	if Options.ServeHTTPS {
-		log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, h))
-	} else {
-		log.Fatal(http.ListenAndServe(address, h))
+	// Interrupt servers on SIGINT/SIGTERM
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
+	// Run listen on http and https ports if iPXE artifacts need to be exposed via HTTP
+	if generateInsecureIPXEURLs {
+		h = app.WithIPXEScriptMiddleware(h)
+	}
```

Comment:
Yeah, this is now being used incorrectly.

In the http only case we don't want this middleware.
This condition should be the old one where we start the middleware only if we're running both servers.

Reply:
Looks like we don't have the same problem in the image service because it's using  the `HasBothHandlers` attribute directly. Not sure why the CAPI tests were complaining then ... will look closer at those.